apps/emqx_oracle/src/emqx_oracle.erl

```diff
@@ -433,9 +433,28 @@ check_if_table_exists(Conn, SQL, Tokens0) ->
     case jamdb_oracle:sql_query(Conn, {SqlQuery, Params}) of
         {ok, [{proc_result, 0, _Description}]} ->
             ok;
-        {ok, [{proc_result, 6550, _Description}]} ->
+        {ok, [{proc_result, 942, _Description}]} ->
             %% Target table is not created
             {error, undefined_table};
+        {ok, [{proc_result, _, Description}]} ->
+            % only the last result is returned, so we need to check on description if it
+            % contains the "Table doesn't exist" error as it can not be the last one.
+            % (for instance, the ORA-06550 can be the result value when table does not exist)
+            case string:find(Description, "ORA-00942: table or view does not exist") of
```

Comment:
I wonder if Oracle databases could have localized error messages, and then this might fail for a non-English localization.  :thinking:

Reply:
Good catch. I'll check for only the ORA code.